### PR TITLE
Add .editorconfig

### PR DESCRIPTION
**Pull request by Goooler**
Let IDEs respect the style.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 4
+indent_style = space
```
Comment (Goooler) [inline .editorconfig:7]: Not sure if we should apply space indent style for all langs.

**Review by RubenRBS — COMMENTED** [review]
Thanks for your contribution! My main concern is similar to the comment I added to the review: How can we handle conflicts between prettier and the editorconfigs in the future?
Either way, this looks like a good idea to have, so thanks :)

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+insert_final_newline = true
+max_line_length = 100
```
[the remark this answers is not in the document]
Comment (Goooler) [reply .editorconfig]: Oooops, this is a copy-paste from another project, removed it for now.

**Comment by partouf** [comment]
`end_of_line` is troublesome for those who have git crlf on auto and are on windows, but not a lot of people actually do develop on windows (although I sometimes do), so maybe it's not so much a practical problem and more a theoretical one

**Pull request merged**